common/src/main/java/com/wynntils/gui/screens/maps/PoiCreationScreen.java

```diff
@@ -0,0 +1,483 @@
+                                // Default to white
+                                colorCache = CommonColors.WHITE;
+                                colorInput.setRenderColor(CommonColors.RED);
+                            } else {
+                                colorCache = color;
+                                colorInput.setRenderColor(CommonColors.GREEN);
+                            }
+
+                            updateSaveStatus();
+                        },
+                        this,
+                        colorInput));
+        if (oldPoi != null && firstSetup) {
+            colorInput.setTextBoxInput(String.valueOf(oldPoi.getColor().toHexString()));
+        } else if (colorInput.getTextBoxInput().isEmpty()) {
+            colorInput.setTextBoxInput("#FFFFFF");
+        }
+        // endregion
+
+        // region Visibility
+        this.addRenderableWidget(
+                new Button(this.width / 2 - 100, this.height / 2 + 90, 20, 20, new TextComponent("<"), (button) -> {
+                    selectedVisiblity = Visibility.values()[
+                            (selectedVisiblity.ordinal() - 1 + Visibility.values().length)
+                                    % Visibility.values().length];
+                }));
+        this.addRenderableWidget(
+                new Button(this.width / 2 + 80, this.height / 2 + 90, 20, 20, new TextComponent(">"), (button) -> {
+                    selectedVisiblity = Visibility.values()[
+                            (selectedVisiblity.ordinal() + 1 + Visibility.values().length)
+                                    % Visibility.values().length];
+                }));
+        if (oldPoi != null && firstSetup) {
+            selectedVisiblity = Visibility.fromMinZoom(oldPoi.getMinZoom());
+        }
+        // endregion
+
+        // region Screen Interactions
+        this.addRenderableWidget(
+                saveButton = new Button(
+                        this.width / 2 + 50,
+                        this.height / 2 + 140,
+                        100,
+                        20,
+                        new TranslatableComponent("screens.wynntils.poiCreation.save"),
+                        (button) -> {
+                            savePoi();
+                            this.onClose();
+                        }));
+
+        this.addRenderableWidget(new Button(
+                this.width / 2 - 150,
+                this.height / 2 + 140,
+                100,
+                20,
+                new TranslatableComponent("screens.wynntils.poiCreation.cancel"),
+                (button) -> this.onClose()));
+        // endregion
+
+        updateSaveStatus();
+        firstSetup = false;
+    }
+
+    @Override
+    public void render(PoseStack poseStack, int mouseX, int mouseY, float partialTick) {
+        super.renderBackground(poseStack);
+        super.render(poseStack, mouseX, mouseY, partialTick);
+
+        FontRenderer.getInstance()
```

Comment:
Is there like no `Label` renderable widget we can use and add to the GUI, instead of having to render this all the time?

Reply:
Even if there is one, this is basically the same computation.